Enable i2s for imx943

biwenli opened the pull request:
Enable i2s for imx943
- sample: i2s_output
- build command: west build -p -b imx943_evk/mimx94398/m33 samples/drivers/i2s/output -d ./build_i2s_output_imx943_evk_m33

LaurentiuM1234 reviewed (commented):
Some DTS-related comments below

biwenli commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_cm.dtsi:
```diff
@@ -65,3 +63,126 @@
+	pinctrl-0 = <&lpi2c6_default>;
+	pinctrl-names = "default";
+	status = "disabled";
+	mux_pca9544a: pca9544a@77 {
```
> add a space before this node's definition, please

sure, np.

biwenli commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.dts:
```diff
@@ -21,66 +22,11 @@
 	};
-};
-
-&emdio {
```
> Are all the ethernet nodes get reused from `imx943_evk_mimx94398_cm.dtsi` now? Could this be a separate commit perhaps, since it doesn't match the `samples: drivers: i2s: output: enable sample for m33 of imx943_evk` well.
> 
> Could you please split that commit up? Thanks

sure, np.

biwenli committed:
boards: nxp: imx943_evk: pinctrl: add sai1_default node

Add sai1_default and lpi2c3_default node

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
boards: imx943_evk: add lpi2c3,lpi2c6, i2s1 node

- Add i2c devices in i2c bus.
- Add i2s node to enable audio samples.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
boards: nxp: imx943_evk: m33: use common dtsi

Use common dtsi imx943_evk_mimx94398_cm.dtsi

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
dts: arm: nxp: imx94x: add i2s, edma1 nodes

Add i2s, edma1 nodes and
reorder flexio3 and flexio4

Signed-off-by: Biwen Li <biwen.li@nxp.com>

JiafeiPan commented:
Please rebase this PR.

uLipe reviewed (dismissed):
Apologies for the delay, LGTM

JiafeiPan reviewed (commented):
Add i2s to supported list in board yaml files.

LaurentiuM1234 reviewed (dismissed):
For the DTS-related changes

biwenli commented on drivers/i2s/i2s_mcux_sai.c:
```diff
@@ -418,6 +418,10 @@ static void enable_mclk_direction(const struct device *dev, bool dir)
+	if (control_base == 0 && offset == 0 && mask == 0) {
+		return;
+	}
+
```
> merge this commit together with the patch "drivers: i2s: mcux_sai: init base,mask,offset as 0" into single one as we need to skip pinmux configuration as no property in the dt.

Sure, np.

biwenli committed:
boards: nxp: imx943_evk: m33: yaml: add i2s

Add i2s for samples/drivers/i2s/output.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

JiafeiPan commented on soc/nxp/imx/imx9/imx943/Kconfig.defconfig.mimx94398.m33:
```diff
@@ -37,4 +37,7 @@ config IDLE_STACK_SIZE
 
 endif
 
+config DMA_TCD_QUEUE_SIZE
```
use configdefault, refer to https://github.com/zephyrproject-rtos/zephyr/pull/98006

JiafeiPan commented:
SAI is also using DMA, right, so do you have this issue on imx943 m-core:
https://github.com/zephyrproject-rtos/zephyr/pull/99661

biwenli commented:
> SAI is also using DMA, right, so do you have this issue on imx943 m-core: #99661

There is no this issue on imx943 m33s. Tested it before the wrong patch.

biwenli committed:
boards: nxp: imx943_evk: disable CONFIG_I2S_HAS_PLL_SETTING

- Disable option CONFIG_I2S_HAS_PLL_SETTING.
- Set CONFIG_I2C_TCA954X_CHANNEL_INIT_PRIO as 51 for
  system to initialize i2c devices properly.

Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_m33.yaml:
```diff
@@ -19,4 +19,5 @@ supported:
   - gpio
```
> Can you please add more details on this. Add i2s for what? what is the topoology of the supported audio scenario.
> 
> how does one uses this to test the functionality.

done

JiafeiPan commented:
Hi, @mmahadevan108 and @dbaluta please review this PR again, thanks.

kartben commented on boards/nxp/imx943_evk/imx943_evk_mimx94398_cm.dtsi:
```diff
@@ -63,3 +63,129 @@
+		status = "disabled";
+
+		mux_i2c@0 {
+			compatible = "ti,tca9548a-channel";
```
this compatible does not exist, nor is it needed to specify a compatible on the child binding. Please drop
